### DISCUSSION TITLE
Configure `dbt_resource` in job config

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#8543] [Q&A] [answered]
Hi everyone!

Is there an option to configure a `dbt_resource` in job config, eg. in dagit?
In documentation I've seen only in code configuration using  `dbt_cli_resource.configured`  method. But this requires to set some configs inside the code.
I want to set `project_dir` and `dbt_executable` outside of code, to have an option to switch between dev and prod envs without chaning code.

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1640680536151300?thread_ts=1640680536.151300&cid=C01U954MEER).*

**Comment by yuhan** [comment, marked as answer]
*Surfacing [an answer](https://dagster.slack.com/archives/C01U954MEER/p1640787807186000?thread_ts=1640680536.151300&cid=C01U954MEER) from Slack:*

In code I've passed `dbt_cli_recource` from `dagster_dbt` directly into `resource_defs` in `to_job` graph method.

my_graph.py
```python
from dagster_dbt import dbt_cli_resource

@op(required_resource_keys={"my_dbt_resource"})
def dbt_run(context):
    context.resources.my_dbt_resource.run()

@graph
def my_graph():
    dbt_run()

my_job = my_graph.to_job(
    resource_defs={
        'my_dbt_resource': dbt_cli_resource
    }
)
```

config yml:
```yaml
resources:
  my_dbt_resource:
    config:
      project_dir: "my/project/dir"
      dbt_executable: "my/dbt/executable"
```
